### DISCUSSION TITLE
missing approximateTerrainHeights.json

### DISCUSSION THREAD

**Discussion by candcconsulting** [iTwin/itwinjs-core#1355] [Q&A] [answered]
We have an app that works in 2.14.3 and can turn on the terrain with the following added to the displayStyle
```
        backgroundMap: {
          useDepthBuffer: false,
          groundBias: -50,
          providerName: "BingProvider",
          providerData: {
            mapType: BackgroundMapType.Hybrid,
          },
          applyTerrain: true,
          transparency: 0.4,
          terrainSettings: {
            providerName: "CesiumWorldTerrain",
            applyLighting: false,
            heightOrigin: -50.0,
            exaggeration: 1.0,
            heightOriginMode: terrainOrigin,
          },
```
However, when we try the same thing with a different 2.14.4 app
We get this error
TypeError: Cannot read property '0-1-0' of null
at ApproximateTerrainHeights.getMinimumMaximumHeights (http://localhost:3000/static/js/0.chunk.js:142938:43)
We have tried copying the approximateTerrainHeights.json into the assets folder, but the same error is displayed
Is there some import or other setting we are missing ... 
Admittedly ... it is for a different project with different iModels / geolocation etc ... so the only difference is not just the iModel.js version
(will try rebuilding with 2.14.3)
Thanks

**Comment by markschlosseratbentley** [comment, marked as answer]
Thanks for the report.

Is the issue-free app (using 2.14.3) a different app from the app using 2.14.4?

In order to narrow this down, could let us know if the issue is still introduced when upgrading from 2.14.3 to 2.14.4 using the exact same app and datasets?

**Reply by calebmshafer** [reply]
From a different, internal, thread with @candcconsulting it should be the same application creating the issue between 2.14.3 and 2.14.4. However we can wait for confirmation there.

@candcconsulting when you did the version upgrade did you run an `npm run clean` or delete the `build/` folder?  If so, you'll want to make sure to run `npm run build` again so that the assets, including `approximateTerrainHeights.json`, are copied over.

The `npm run start` script, technically `react-scripts start`, does not currently copy the assets over, however there is a open PR to change that https://github.com/imodeljs/create-react-app/pull/37.

**Reply by candcconsulting** [reply]
It is two different apps ... I am trying to rebuild with 2.14.3 but I think it could be related to the npm run build ...

**Comment by candcconsulting** [comment]
Thanks Caleb, I reset back to 2.14.4 and rebuilt my nodes_module and rebuilt with npm run build ... and it also resolved some issues we were having with viewing the reality models also
![image](https://user-images.githubusercontent.com/19268085/117472893-49c82f80-af51-11eb-8052-a30862ebdac2.png)
